s3readerat.go

```diff
@@ -163,3 +202,98 @@ func (ra *S3ReaderAt) ReadAt(p []byte, off int64) (n int, err error) {
 
 	return n, err
 }
+
+func (ra *S3ReaderAt) s3Client() *s3.Client {
+	if ra.client != nil {
+		return ra.client
+	}
+
+	if ra.options == nil {
+		return nil
+	}
+
+	ra.client = s3.New(*ra.options)
+
+	return ra.client
+}
+
+func (ra *S3ReaderAt) s3ClientInRegion(region string) *s3.Client {
+	// Single-region mode.
+	if ra.options == nil {
+		return nil
+	}
+
+	// Multi-region mode. Already have s3.Client.
+	if ra.options.Region == region && ra.client != nil {
+		return ra.client
+	}
+
+	// Multi-region mode. Need a new s3.Client.
+	options := *ra.options
+	options.Region = region
```

Comment:
we need to test to make sure we don't override the current `ra.options` and just create a copy